src/test/codegen/repr-transparent.rs

```diff
@@ -138,22 +138,6 @@ pub extern fn test_Projection(_: StructWithProjection) -> StructWithProjection {
 // gotten rid of pointee types yet, the IR function signature will be syntactically different (%Foo*
 // vs %FooWrapper*).
 
-#[repr(C)]
-pub struct Rgb8 { r: u8, g: u8, b: u8 }
-
-#[repr(transparent)]
-pub struct Rgb8Wrap(Rgb8);
-
-// NB: closing parenthesis is missing because sometimes the argument has a name and sometimes not
-// CHECK: define i32 @test_Rgb8Wrap(i32
-#[no_mangle]
-#[cfg(all(target_arch="x86_64", target_os="linux"))]
-pub extern fn test_Rgb8Wrap(_: Rgb8Wrap) -> Rgb8Wrap { loop {} }
-
-#[cfg(not(all(target_arch="x86_64", target_os="linux")))]
-#[no_mangle]
-pub extern fn test_Rgb8Wrap(_: u32) -> u32 { loop {} }
-
 // Same as with the small struct above: ABI-dependent, we only test the interesting case
```

Comment:
This comment and the one above doesn't quite fit any more. I'd move the `union FloatBits` test below into `repr-transparent-x86_64.rs` (perhaps better called `repr-transparent-sysv64.rs`?) and replace everything from line 126 onwards with:

```
 // All that remains to be tested are aggregates. They are tested in separate files called repr-
 // transparent-aggregate-*.rs  with `only-*` or `ignore-*` directives, because the expected LLVM IR
 // function signatures vary so much that it's not reasonably possible to cover all of them with a
 // single CHECK line.
 //
 // You may be wondering why we don't just compare the return types and argument types for equality
 // with FileCheck regex captures. Well, rustc doesn't perform newtype unwrapping on newtypes
 // containing aggregates. This is OK on all ABIs we support, but because LLVM has not gotten rid of
 // pointee types yet, the IR function signature will be syntactically different (%Foo* vs
 // %FooWrapper*).
```